src/test/java/pl/net/was/TestOpenApiQueries.java

```diff
@@ -124,4 +125,53 @@ public void selectItems()
                 null,
                 List.of(LocalDate.of(2007, 10, 10), LocalDate.of(2022, 12, 8)));
     }
+
+    @Test
+    public void searchItemsWithInPhrase()
+    {
+        List<MaterializedRow> rows = getQueryRunner().execute("SELECT name FROM fastapi.default.search WHERE item_ids IN (ARRAY['2'])").getMaterializedRows();
+        assertThat(rows).size().isEqualTo(1);
+        assertThat(rows.getFirst().getFields()).first().isEqualTo("Plumbus");
+
+        rows = getQueryRunner().execute("SELECT name FROM fastapi.default.search WHERE item_ids IN (ARRAY['1', '2'])").getMaterializedRows();
+        assertThat(rows).size().isEqualTo(2);
+    }
+
+    @Test
+    public void searchItemsWithSubQuery10k()
+    {
+        generateDataset("memory.default.test_items_10k", 10000);
+        List<MaterializedRow> rows = getQueryRunner().execute("SELECT name FROM fastapi.default.search WHERE item_ids IN (select array_agg(item_id) from memory.default.test_items_10k)").getMaterializedRows();
+        assertThat(rows)
+                .extracting(row -> row.getFields().getFirst())
+                .containsExactly("Portal Gun", "Plumbus");
+    }
+
+    @Test
+    public void searchItemsWithSubQuery100k()
+    {
+        generateDataset("memory.default.test_items_100k", 100000);
+        List<MaterializedRow> rows = getQueryRunner().execute("SELECT name FROM fastapi.default.search WHERE item_ids IN (select array_agg(item_id) from memory.default.test_items_100k)").getMaterializedRows();
+        assertThat(rows)
+                .extracting(row -> row.getFields().getFirst())
+                .containsExactly("Portal Gun", "Plumbus");
+    }
+
+    private void generateDataset(String name, int elements)
+    {
+        getQueryRunner().execute("CREATE TABLE IF NOT EXISTS %s ( item_id VARCHAR(255) )".formatted(name));
+
+        int batchSize = 5000; //limit of the SEQUENCE
+        int batches = elements / batchSize;
+        String query = """
+                       INSERT INTO %s
+                       WITH numbers AS (
+                         SELECT SEQUENCE(%d, %d) AS id_range
+                       )
+                       SELECT CAST(id AS VARCHAR) AS item_id FROM numbers CROSS JOIN UNNEST(id_range) AS t(id)
+                       """;
+
+        IntStream.range(0, batches).forEach(i -> getQueryRunner().execute(query.formatted(name, i * batchSize, i * batchSize + batchSize)));
```

Comment:
Use the `sequence` table function, which doesn't have the 10k limit. With this, since it's a single query, it's better to inline this, so it's more clear that the generated dataset is a table with sequential numbers. It would be also nice to clean up these tables once the test is finished, consider using https://github.com/trinodb/trino/blob/master/testing/trino-testing/src/main/java/io/trino/testing/sql/TestTable.java#L30 in a `try` block.
```suggestion
        getQueryRunner().execute("CREATE TABLE %s AS SELECT CAST(sequential_number AS VARCHAR) AS item_id FROM TABLE(sequence(start=>0, stop=>%d))".formatted(name, elements-1));
```

Reply:
Done, thanks for the suggestion